pennylane/templates/subroutines/qsvt.py

```diff
@@ -150,6 +150,201 @@ def qsvt(A, angles, wires, convention=None):
     return QSVT(UA, projectors)
 
 
+# pylint: disable=too-many-branches
+def qsvt(A, poly, encoding_wires, block_encoding=None):
+    r"""
+    Implements the Quantum Singular Value Transformation (QSVT) for a matrix or Hamiltonian ``A``, using a polynomial
+    defined by ``poly`` and a block encoding specified by ``block_encoding``.
+
+    .. math::
+
+        \begin{pmatrix}
+        A & * \\
+        * & * \\
+        \end{pmatrix}
+        \xrightarrow{QSVT}
+        \begin{pmatrix}
+        \text{poly}(A) + i \dots & * \\
+        * & * \\
+        \end{pmatrix}
+
+    The polynomial transformation is encoded as the real part of the top left term after applying the operator.
+
+    This function calculates the required phase angles from the polynomial using :func:`~.poly_to_angles`.
+
+    .. note::
+
+        The function ``poly_to_angles``, used within ``qsvt``, is not JIT-compatible, which prevents ``poly`` from being
+        traceable in ``qsvt``. However, ``A`` is traceable and can be optimized by JIT within this function.
+
+    Args:
+
+        A (Union[tensor_like, Operator]): The matrix on which the QSVT will be applied.
+            This can be an array or an object that has a Pauli representation.
+
+        poly (tensor_like): Polynomial coefficients defining the transformation, represented in increasing order of degree.
+            This means the first coefficient corresponds to the constant term, the second to the linear term, and so on.
+
+        encoding_wires (Sequence[int]): The qubit wires used for the block encoding. See Usage Details below for
+            more information on ``encoding_wires`` depending on the block encoding used.
+
+        block_encoding (str): Specifies the type of block encoding to use. Options include:
+
+            - ``"prepselprep"``: Embeds the Hamiltonian ``A`` using :class:`~pennylane.PrepSelPrep`. Default encoding for Hamiltonians.
+            - ``"qubitization"``: Embeds the Hamiltonian ``A`` using :class:`~pennylane.Qubitization`.
+            - ``"embedding"``: Embeds the matrix ``A`` using :class:`~pennylane.BlockEncode`. Template not hardware compatible.
+            - ``"fable"``: Embeds the matrix ``A`` using :class:`~pennylane.FABLE`. Default encoding for matrices.
```

Comment:
Please bring this line before embedding, so the user sees the default option first.